pkg/controller/templateupdaterequest/templateupdaterequest_controller.go

```diff
@@ -0,0 +1,324 @@
+package templateupdaterequest
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/host-operator/pkg/configuration"
+	"github.com/codeready-toolchain/host-operator/pkg/controller/nstemplatetier"
+	"github.com/codeready-toolchain/toolchain-common/pkg/condition"
+	"github.com/go-logr/logr"
+
+	errs "github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var log = logf.Log.WithName("controller_templateupdaterequest")
+
+const (
+	// DeletionTimeout the duration after which the TemplateUpdateRequest can be deleted, once it reached the `complete=true` condition
+	DeletionTimeout = 5 * time.Second
```

Comment:
this value will be configurable in the future: see https://issues.redhat.com/browse/CRT-707